src/libraries/System.Net.Ping/src/System/Net/NetworkInformation/Ping.PingUtility.cs

```diff
@@ -90,15 +90,30 @@ private async Task<PingReply> SendWithPingUtilityAsync(IPAddress address, byte[]
                     return CreatePingReply(IPStatus.TimedOut);
                 }
 
-                if (p.ExitCode == 1 || p.ExitCode == 2)
+                try
                 {
+                    string? output = null;
                     // Throw timeout for known failure return codes from ping functions.
-                    return CreatePingReply(IPStatus.TimedOut);
-                }
+                    if (p.ExitCode == 1 || p.ExitCode == 2)
+                    {
+                        if (options?.Ttl != null)
+                        {
+                            // TTL exceeded may have occured
+                            output = await p.StandardOutput.ReadToEndAsync().ConfigureAwait(false);
+                            if (output.Contains("Time to live exceeded", StringComparison.Ordinal))
```

Comment:
This is going to be problematic IMHO. This message may come out localized. I'm wondering if we should accept that we cannot really distinguish the cases with confidence..

cc: @filipnavara @tmds

Reply:
maybe not. It seems like we explicitly set LC_ALL but dependency on particular text string still worries me little bit.

Reply:
IMHO it's worth a try. We can check strings in the common implementations (FreeBSD, GNU Inetutils, macOS*). Worst case it will behave the same way as it does now, best case we can provide slightly more accurate result.

However, the code needs to do way more input checking than it does now. The `.IndexOf` calls may fail to find anything and it would throw `ArgumentOutOfRangeException`.

Reply:
I realize that the dependency is suboptimal, one of the reason I put up this PR is to check is to check if it gets a hit on CI (I tried both Arch and Ubuntu locally, so I have hopes)

Will add checking of the indices